agent/tool-scripts/base-tool

```diff
@@ -578,15 +578,15 @@ install)
 	function require_rpm {
 		_tpn="${1}"
 		_tpv="${2}"
-		require-rpm ${_tpn} ${_tpv}
+		_installed_rpm="$(require-rpm "${_tpn}" "${_tpv}")"
```

Comment:
The original code invoked `require-rpm` without ensuring the version string was passed along, since an empty `_tpv` variable would not be expanded.

In the new form, `require-rpm` needs to be sure that a version string is passed if a match argument is provided.  So by quoting the argument variables it gets passed along as is, that is, as an empty string.

While we don't support range versions of tools today (we can't with `sysstat` they break compatibility in their output formats), I wanted this code to match what we are doing elsewhere.

Reply:
Ah...then I believe that there is a problem here:  your intent was to have nested quotes, but, I think, you have the opposite!  In your proposed code, there are no quotes around the expansion of the `_tpv` variable -- the mark preceding it is a _closing quote_ and the mark following it is an _opening quote_.

I don't see any need for the enclosing quotes around the "command substitution", so rather than my suggestion above, perhaps this would be appropriate:
```
		_installed_rpm=$(require-rpm "${_tpn}" "${_tpv}")
```
(If you actually do need the outer quotes, then I think you need to escape the inner ones, or you have to go with an even more cumbersome approach.)

Reply:
You need the quotation marks around the individual variables being referenced so that they are expanded properly as that argument number.

In this case it doesn't matter, but it would when you add a 3rd argument (should be support minimum versions).

E.g.:
```
$ cat foo-bar
#!/bin/bash
echo "1: '${1}'; 2: '${2}'; 3: '${3}'"
$ cat testit
#!/bin/bash
x="$(./foo-bar "${1}" " " "${2}")"
y=$(./foo-bar "${1}" " " "${2}")
z="$(./foo-bar ${1} ${2})"

echo "x: (${x})"
echo "y: (${y})"
echo "z: (${z})"
$ ./testit a b
x: (1: 'a'; 2: ' '; 3: 'b')
y: (1: 'a'; 2: ' '; 3: 'b')
z: (1: 'a'; 2: 'b'; 3: '')
```

The extra double-quotes are needed to make sure the 3rd parameter is `b`, but the surrounding quotes are not considered when interpreting inside the `$()` construct.

Reply:
> the surrounding quotes are not considered when interpreting inside the $() construct.

Thanks for the education!  However, I'm going to go ahead and say that this is a botch -- it's confusing to read and therefore should be avoided.

I continue to recommend my previous suggestion:  omit the outer quotes.  In your example, I see no difference between your "x" case and your "y" case, which I think indicates that the outer quotes are unnecessary; removing them removes the ambiguity/confusion for the reader.

Reply:
Okay, I'll drop the outer quotes if that is confusing to you.  We have been using them throughout the code for a while.